File: backend/src/main/java/com/example/backend/model/Leave.java
package com.example.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Leave {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int emergencyContact;
    private Date startDate;
    private Date endDate;
    private int durationOfLeave;
    private String causeOfLeave;
    private Boolean leaveStatus;

    public Leave(Long id) {
        this.id = id;
    }

}

File: frontend/src/app/dto/Leave.ts
import { Employee } from "./Employee";
import { LeaveType } from "./LeaveType";

export class Leave {

  id: number | null = null;
  leaveType: LeaveType | null = null;
  employee: Employee | null = null;
  emergencyContact: string | null = null;
  startDate: Date | null = null;
  endDate: Date | null = null;
  durationOfLeave: number | null = null;
  causeOfLeave: string | null = null;
  leaveStatus: boolean | null = null;
  approvedBy: Employee | null = null;
  
  constructor(o?: Partial<Leave>) {
    Object.assign(this, o);
  }
}
